commerce/domain/checkout/giftcard/GiftCardAdapterSample.cls

```diff
@@ -0,0 +1,109 @@
+        String giftCardCurrency = applyRequest.getCurrencyIsoCode();
+        String giftCardCode = applyRequest.getGiftCardCode();
+        String pin = applyRequest.getPin();
+
+        CartExtension.ApplyGiftCardResponse response = null;
+
+        if(useHTTPService) {
+           response = applyGiftCardService(giftCardCode, pin, amount, giftCardCurrency);
+        } else {
+           response = applyGiftCardMockedService(giftCardCode, pin, amount, giftCardCurrency);
+        }
+
+        return response;
+
+    }
+
+    private CartExtension.ApplyGiftCardResponse applyGiftCardMockedService(String giftCardCode, String giftCardPin,
+                                                Double amount, String giftCardCurrency) {
+        CartExtension.ApplyGiftCardResponse applyResponse = new CartExtension.ApplyGiftCardResponse();
+
+        applyResponse.setGiftCardReference('123456789');
+        applyResponse.setAppliedAmount(50);
+        applyResponse.setLastFour('6789');
+        applyResponse.setCreditType('GiftCard');
+        applyResponse.setStatus('Success');
+
+        return applyResponse;
+
+
+    }
+
+   private CartExtension.ApplyGiftCardResponse applyGiftCardService(String giftCardCode, String giftCardPin,
+                                               Double amount, String giftCardCurrency) {
+        CartExtension.ApplyGiftCardResponse applyResponse = new CartExtension.ApplyGiftCardResponse();
+
+         final Integer successfulHttpRequest = 200;
+
+        Http http = new Http();
+        HttpRequest request = new HttpRequest();
+        request.setEndpoint(httpHost + '/apply-gift-card');
+        request.setMethod('POST');
+        HttpResponse response = http.send(request);
+
+
+        if (response.getStatusCode() == successfulHttpRequest) {
+               // Parse JSON response
+                String responseBody = response.getBody();
+
+                // Parse JSON using JSON.deserialize or JSON.deserializeUntyped
+                Map<String, Object> resultMap = (Map<String, Object>) JSON.deserializeUntyped(responseBody);
+
+               String giftCardReference = (String)resultMap.get('giftCardReference');
+
+               Double availableAmount = (Double)resultMap.get('availableAmount');
+               if(availableAmount >= amount){
+                    applyResponse.setAppliedAmount(amount);
+                } else if(availableAmount < amount && availableAmount>0){
+                    applyResponse.setAppliedAmount(availableAmount);
+                } else if(availableAmount == 0){
+                    applyResponse.setAppliedAmount(0);
+                    applyResponse.setErrorMessage('Zero balance in gift card');
+                    applyResponse.setStatus('Fail');
+                    applyResponse.setGiftCardReference(giftCardReference);
+                    applyResponse.setLastFour(giftCardReference.substring(giftCardReference.length() - 4));
+                    applyResponse.setCreditType('GiftCard');
```

Comment:
Can we set it within the Java Service itself so that we can map it to the Enum value like @gurpreetsainisalesforce mentioned